core/chains/evm/txmgr/txmgr_test.go

```diff
@@ -32,8 +34,38 @@ import (
 	"github.com/smartcontractkit/chainlink/core/services/pg"
 	pgmocks "github.com/smartcontractkit/chainlink/core/services/pg/mocks"
 	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/smartcontractkit/sqlx"
 )
 
+func makeTestEvmTxm(t *testing.T, db *sqlx.DB, ethClient evmclient.Client, cfg txmgr.Config, keyStore txmgr.KeyStore, eventBroadcaster pg.EventBroadcaster) txmgr.TxManager {
+	lggr := logger.TestLogger(t)
+	checkerFactory := &testCheckerFactory{}
+	lp := logpoller.NewLogPoller(logpoller.NewORM(testutils.FixtureChainID, db, lggr, pgtest.NewQConfig(true)), ethClient, lggr, 100*time.Millisecond, 2, 3, 2, 1000)
+
+	// logic for building components (from evm/evm_txm.go) -------
+	lggr.Infow("Initializing EVM transaction manager",
+		"gasBumpTxDepth", cfg.EvmGasBumpTxDepth(),
+		"maxInFlightTransactions", cfg.EvmMaxInFlightTransactions(),
+		"maxQueuedTransactions", cfg.EvmMaxQueuedTransactions(),
+		"nonceAutoSync", cfg.EvmNonceAutoSync(),
+		"gasLimitDefault", cfg.EvmGasLimitDefault(),
+	)
+
+	// build estimator from factory
+	estimator := gas.NewEstimator(lggr, ethClient, cfg)
+
+	// build forwarder manager for evm txm
+	var fwdMgr *forwarders.FwdMgr
+	if cfg.EvmUseForwarders() {
+		fwdMgr = forwarders.NewFwdMgr(db, ethClient, lp, lggr, cfg)
+	} else {
+		lggr.Info("EvmForwarderManager: Disabled")
+	}
+	// --------------------
+
+	return txmgr.NewTxm(db, ethClient, cfg, keyStore, eventBroadcaster, lggr, checkerFactory, estimator, fwdMgr)
+}
+
```

Comment:
so this function exists to keep the test cases in this file working. but it is a near duplicate of `newEvmTxm`.
this solution keeps the current test file where it is while we work on moving pieces out of the transaction manager

the other solution is to move this file and have it test directly against `newEvmTxm`. thoughts?

Reply:
also note: this will eventually have to move to test `func newEvmTxm(...)` anyway. It's mostly a matter of when to move it.

Reply:
One option is define the helper function in `helpers_test.go` while reusing all the `newEvmTxm` logic like so

helpers_test.go
```
func makeTestEvmTxm (...) {
...
newEvmTxm(...)
```

and import makeTestEvmTxm here into this file

Reply:
that could work - I think I just don't want it to be widely used, since it will be removed in the future anyway